### ISSUE TITLE
SC 2.5.8 Target Size (Minimum) - Is "24 by 24 CSS pixels” reasonable?

### ISSUE THREAD

**Issue by caztcha** [w3c/wcag#1831]
On the 2021-05-21 working draft, the SC 2.5.8 is renamed as "Target Size (Minimum)”, and the size requirement is reduce to "24 by 24 CSS pixels". What is the basis for this "24 by 24 CSS pixels" ?

I think the "44 by 44 CSS pixels" standard described in the already-existing SC 2.5.5 is reasonable. The Apple Human Interface Guidelines say, 

> Provide ample touch targets for interactive elements. Try to maintain a minimum tappable area of 44pt x 44pt for all controls.
(https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/)

Also, the Google Material Design document say,

> For most platforms, consider making touch targets at least 48 x 48 dp. A touch target of this size results in a physical size of about 9mm, regardless of screen size. The recommended target size for touchscreen elements is 7-10mm. It may be appropriate to use larger touch targets to accommodate a larger spectrum of users.
(https://material.io/design/usability/accessibility.html#layout-and-typography)

On the other hand, there does not seem to be any validity to the "24 by 24 CSS pixels” described in the SC 2.5.8.

So, my suggestion is:

- Remove the SC 2.5.8
- Make the SC 2.5.5 (named "Target Size" in WCAG 2.1) to be the AA level. 

By doing so, the unnatural order of the SCs ("Target Size (Enhanced)" is before "Target Size (Minimum)") will also be resolved as a result.

The "Understanding" draft for the SC 2.5.8 ( https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum ) has examples of "dropdown menus" and "stacked links (numbered lists)", but I think these can be considered as a form of the "block of text", so I think they can be exceptions to the SC 2.5.5.

**Comment by patrickhlauke** [comment]
Two points worth noting:

* Apple's and Google's guidelines are specifically about touch. WCAG's target SCs are scoped to pointers in general, i.e. also mouse.  this influenced the decision to go for a more realistic/smaller minimum sizing, as it would be difficult to ask that sites all make their targets touch-friendly-sized by default (the vast majority of content on the web today would immediately be non-compliant) - which is where a compromise of 24x24 was then worked out. Admittedly it's a bit more arbitrary than the 44x44 which was inspired/taken by the AHI - it's essentially quarter the size/area of Google's target size.
* the proposed 2.5.8 expands the concept to not just look at the actual target size itself, but to also take into consideration any "clearance" between the target and neighboring targets - this concept it absent from 2.5.5

**Comment by caztcha** [comment]
Thank you for your comment. I see that the new SC 2.5.8 scopes the pointers in general, including the mouse (not just the touch interface).

If this SC 2.5.8 is to remain, it would be nice to have a statement that this SC is primarily a requirement for mice. This is to avoid misunderstandings like mine that WCAG adopts a looser standard than Apple HIG’s 44 by 44.

**Comment by mraccess77** [comment]
It is my personal understanding that Apple's 44x44 pixel target is not a standard but a guideline/recommendation and one that they don't always follow themselves.  It's geared toward mobile apps while WCAG is aimed at web content more broadly.  Because of this and the difficulty of getting a WCAG A/AA criteria for 44x44 allowed this new criteria to be accept and benefit folks even though it is understood that will not assist as many people as the 44x44 one.

**Comment by patrickhlauke** [comment]
> I see that the new SC 2.5.8 scopes the pointers in general

for completeness, 2.5.5 is also scoped to pointers, and actually uses "pointer inputs" in its normative text.

> If this SC 2.5.8 is to remain, it would be nice to have a statement that this SC is primarily a requirement for mice.

it's a requirement for all targets in general, not specifically for mice or touch or anything. it does however use "target" which is linked to the definition that includes mention of pointers in general https://www.w3.org/TR/WCAG22/#dfn-target

**Comment by caztcha** [comment]
Thank you Patrick, Jonathan, for your considering and replying for my questions and opinions. Now I could understand the positioning of the new SC 2.5.8.

**Comment by jenstrickland** [comment]
Draft proposed response:
<hr>

SC 2.5.8 Target Size (Minimum) (Level AA) is a partner to SC 2.5.5 Target Size (Level AAA).

- The minimum target size to meet Level AA is 24x24 pixels and includes a target offset of at least 24 pixels, with noted exceptions. There are some interfaces where smaller sizes are appropriate and necessary, for example the toolbars at the top of certain editing interfaces. However, this SC should encourage the move towards larger target sizes and offset spacing in redesign efforts, as appropriate. 

- This SC 2.5.8 includes the range of pointer possibilities, from mouse to touch, with the mobility needs as a consideration. 

- The SC 2.5.5 Target Size (Level AAA) may be perceived as an enhancement increasing target size to 44x44 pixels where that is appropriate to the interface.

**Comment by alastc** [comment]
The group accepted the response above.

I'll also put, for the record, a little extra info from @DavidMacDonald:

<hr> 

Apple's 44px guidelines and Google's 48 px are for (finger) touch interfaces whereas WCAG applies to all types of input (touch, mouse, pen, etc.) It should be noted that many Apple and Google interface components don't meet their own size guidelines for various reasons. We have to be careful with Success Criteria because in many jurisdictions they will become law, not guidelines that can be broken when its appropriate.

There are studies that indicate many people with motor disabilities require 100px targets, in which case the Google and Apple guidance would not be sufficient.

This Success Criteria has been through several years of iterations. It provides a careful balance between the impact on Web design and benefit to users. Requirements that impact the visual design of web sites typically require more compromise.

This success criterion is basically saying: "Don't make really small targets". It should be noted that users can obtain further increases to target size size by zooming in when necessary.

The SC 2.5.5 Target Size (Level AAA) is an enhancement increasing target size to 44x44 pixels where that is appropriate to the interface.
